packages/react/src/__tests__/ReactElementJSX-test.js

```diff
@@ -48,23 +52,25 @@ describe('ReactElement.jsx', () => {
     expect(element.constructor).toBe(object.constructor);
   });
 
-  it('should use default prop value when removing a prop', () => {
+  it('should use default prop value when removing a prop', async () => {
     class Component extends React.Component {
       render() {
-        return JSXRuntime.jsx('span', {});
+        return JSXRuntime.jsx('span', {children: [this.props.fruit]});
       }
     }
     Component.defaultProps = {fruit: 'persimmon'};
 
     const container = document.createElement('div');
-    const instance = ReactDOM.render(
-      JSXRuntime.jsx(Component, {fruit: 'mango'}),
-      container,
-    );
-    expect(instance.props.fruit).toBe('mango');
+    const root = ReactDOMClient.createRoot(container);
+    await act(() => {
+      root.render(JSXRuntime.jsx(Component, {fruit: 'mango'}));
+    });
+    expect(container.firstChild.textContent).toBe('mango');
```

Comment:
Switched this to use textContent since we don't have access to the instance in createRoot.

Reply:
What I've done sometimes is to use a ref, but this is actually cleaner.